### ISSUE TITLE
Albert xLarge pretraining idles on single v3-8 tpu

### ISSUE THREAD

**Issue by dhruvsakalley** [google-research/google-research#146]
I'm trying to train with ~ 568 tfrecord files totaling upto 1.9 TB, created from ~30GB of text and generated using 128 max_seq_length and custom vocab comparable to the original vocab size.

Here are my training flasgs:
nohup python3 run_pretraining.py \
  --input_file=gs://databucket-* \
  --output_dir=gs://output bucket \
  --export_dir==gs://export bucket \
  --do_train=True \
  --do_eval=True \
  --albert_config_file=gs://<config bucket>\albert_config.json \
  --train_batch_size=512 \
  --eval_batch_size=64 \
  --max_seq_length=128 \
  --max_predictions_per_seq=20 \
  --num_train_steps=2000000 \
  --num_warmup_steps=10000 \
  --learning_rate=1e-4 \
  --use_tpu=True \
  --tpu_name=gorilla-alb-tpu \
  --save_checkpoints_steps=10000 \
  --max_eval_steps=100 \
  --num_tpu_cores=8 \
  --user_tpu=True \
  --keep_checkpoint_max=500 \
  \> iter001.log &

I tried this on the preemptible tpu first and did not even get one checkpoint, but later on the second iteration, I used a dedicated TPU, but after the global step 19000 its started idling and did not produce any checkpoints, as can be seen in the logs below. 
Q1: Any idea what am I missing here? 
Q2: Also, why is the train time so slow compared to BERT, which stepped through 10K/20 mins, here it seems 10K/1.15 hr or more?
Q3: What is the recommended batch size for single TPU instance and how does it impact the train steps and the learning rate?

Extract of the logs showing idling here : 
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:04:59.069434 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
INFO:tensorflow:loss = 6.453083, step = 19000 (1816.160 sec)
I1124 07:05:04.021676 139980433229568 basic_session_run_hooks.py:260] loss = 6.453083, step = 19000 (1816.160 sec)
INFO:tensorflow:global_step/sec: 0.550612
I1124 07:05:04.024546 139980433229568 tpu_estimator.py:2307] global_step/sec: 0.550612
INFO:tensorflow:examples/sec: 281.913
I1124 07:05:04.024918 139980433229568 tpu_estimator.py:2308] examples/sec: 281.913
INFO:tensorflow:Enqueue next (1000) batch(es) of data to infeed.
I1124 07:05:04.025995 139980433229568 tpu_estimator.py:600] Enqueue next (1000) batch(es) of data to infeed.
INFO:tensorflow:Dequeue next (1000) batch(es) of data from outfeed.
I1124 07:05:04.026226 139980433229568 tpu_estimator.py:604] Dequeue next (1000) batch(es) of data from outfeed.
INFO:tensorflow:Outfeed finished for iteration (19, 12)
I1124 07:05:28.020126 139978655139584 tpu_estimator.py:279] Outfeed finished for iteration (19, 12)
I1124 07:05:29.137331 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:05:29.192949 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
I1124 07:05:59.262590 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:05:59.311209 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
I1124 07:06:29.373492 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:06:29.447152 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
INFO:tensorflow:Outfeed finished for iteration (19, 46)
I1124 07:06:29.709674 139978655139584 tpu_estimator.py:279] Outfeed finished for iteration (19, 46)
I1124 07:06:59.522179 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:06:59.572463 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
I1124 07:07:29.611606 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:07:29.668322 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
INFO:tensorflow:Outfeed finished for iteration (19, 80)
I1124 07:07:31.398825 139978655139584 tpu_estimator.py:279] Outfeed finished for iteration (19, 80)
I1124 07:07:59.732999 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:07:59.793257 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
I1124 07:08:29.856629 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:08:29.919902 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
INFO:tensorflow:Outfeed finished for iteration (19, 114)
I1124 07:08:33.089343 139978655139584 tpu_estimator.py:279] Outfeed finished for iteration (19, 114)
I1124 07:08:59.989946 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:09:00.072418 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
I1124 07:09:30.138386 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:09:30.192759 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
INFO:tensorflow:Outfeed finished for iteration (19, 148)
I1124 07:09:34.782029 139978655139584 tpu_estimator.py:279] Outfeed finished for iteration (19, 148)
I1124 07:10:00.262878 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:10:00.322918 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
I1124 07:10:30.385881 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:10:30.438601 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
INFO:tensorflow:Outfeed finished for iteration (19, 182)
I1124 07:10:36.469368 139978655139584 tpu_estimator.py:279] Outfeed finished for iteration (19, 182)
I1124 07:11:00.488457 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:11:00.554763 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
I1124 07:11:30.607665 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:11:30.657622 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
INFO:tensorflow:Outfeed finished for iteration (19, 216)
I1124 07:11:38.161521 139978655139584 tpu_estimator.py:279] Outfeed finished for iteration (19, 216)
I1124 07:12:00.723495 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:12:00.779349 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
I1124 07:12:30.844133 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:12:30.897756 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
INFO:tensorflow:Outfeed finished for iteration (19, 250)
I1124 07:12:39.851271 139978655139584 tpu_estimator.py:279] Outfeed finished for iteration (19, 250)
I1124 07:13:00.941801 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:13:00.998180 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
I1124 07:13:31.063151 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:13:31.136875 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
INFO:tensorflow:Outfeed finished for iteration (19, 284)
I1124 07:13:41.539725 139978655139584 tpu_estimator.py:279] Outfeed finished for iteration (19, 284)
I1124 07:14:01.204144 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:14:01.270991 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
I1124 07:14:31.314057 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:14:31.368036 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
INFO:tensorflow:Outfeed finished for iteration (19, 318)
I1124 07:14:43.230399 139978655139584 tpu_estimator.py:279] Outfeed finished for iteration (19, 318)
I1124 07:15:01.408405 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:15:01.466477 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
I1124 07:15:31.531250 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:15:31.588218 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
INFO:tensorflow:Outfeed finished for iteration (19, 352)
I1124 07:15:44.920365 139978655139584 tpu_estimator.py:279] Outfeed finished for iteration (19, 352)
I1124 07:16:01.650633 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:16:01.709294 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
I1124 07:16:31.747833 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:16:31.803840 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
INFO:tensorflow:Outfeed finished for iteration (19, 386)
I1124 07:16:46.611431 139978655139584 tpu_estimator.py:279] Outfeed finished for iteration (19, 386)
I1124 07:17:01.843595 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:17:01.905045 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
I1124 07:17:31.969744 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:17:32.024375 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
INFO:tensorflow:Outfeed finished for iteration (19, 420)
I1124 07:17:48.304604 139978655139584 tpu_estimator.py:279] Outfeed finished for iteration (19, 420)
I1124 07:18:02.091745 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:18:02.247168 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
I1124 07:18:32.311644 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:18:32.362884 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
I1124 07:19:02.426864 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:19:02.486614 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
I1124 07:19:32.542831 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:19:32.602440 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
I1124 07:20:02.665383 139978646746880 transport.py:157] Attempting refresh to obtain initial access_token
WARNING:tensorflow:TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.
W1124 07:20:02.727149 139978646746880 preempted_hook.py:91] TPUPollingThread found TPU b'gorilla-alb-tpu' in state READY, and health HEALTHY.

[iter001.log](https://github.com/google-research/google-research/files/3883630/iter001.log)

**Comment by josecannete** [comment]
Same problem, also here: https://github.com/tensorflow/tpu/issues/609

**Comment by pwais** [comment]
@dhruvsakalley @josecannete Google Cloud has had a bug internally for about a month.  They just pushed a fix.  I had to delete and re-recreate my TPUs, but now I see uninterrupted training.  It's unclear if your issue is due to the same bug, but you might try re-creating your TPUs.  (Don't just stop-start them, you have to delete them apparently).  https://github.com/tensorflow/tpu/issues/609#issuecomment-558819717

**Comment by mananeau** [comment]
Hi all. I am facing the same issue, tried deleting and re-creating the TPUs but it didn't solve the problem. @dhruvsakalley @josecannete would you mind telling me how you solved the problem?

**Comment by dhruvsakalley** [comment]
After the bugfix on the TPU VM I didn't encounter the issue again.

**Comment by josecannete** [comment]
Same for me. The problem was not in our part.
